test/integration/generate-ignores.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const glob = require('glob');
+const fs = require('fs');
+const js = {};
+const native = {};
+
+glob("*-tests/**/*.json", (er, files) => {
+    for (const file of files) {
+        const style = require(`./${file}`);
+        const key = file.replace(/(.*)\/style\.json/, '$1');
+
+        if (style.metadata && style.metadata.test) {
+            if (style.metadata.test.ignored) {
+                if (style.metadata.test.ignored.native) {
+                    native[key] = style.metadata.test.ignored.native;
+                }
+                if (style.metadata.test.ignored.js) {
+                    js[key] = style.metadata.test.ignored.js;
+                }
+            }
+
+            if (style.metadata.test.skipped) {
+                if (style.metadata.test.skipped.native) {
+                    native[key] = `skip - ${style.metadata.test.skipped.native}`;
+                }
+                if (style.metadata.test.skipped.js) {
+                    js[key] = `skip - ${style.metadata.test.skipped.js}`;
+                }
+            }
+
+            if (style.metadata.test.ignored || style.metadata.test.skipped) {
+                delete style.metadata.test.ignored;
+                delete style.metadata.test.skipped;
+                fs.writeFileSync(file, JSON.stringify(style, null, 2));
+            }
+        }
+    }
+
+    fs.writeFileSync('ignores-native.json', JSON.stringify(native, null, 2));
+    fs.writeFileSync('ignores-js.json', JSON.stringify(js, null, 2));
+});
```

Comment:
I might need to run it again if the files changes before this PR merges, but I'll remove it before merging.